docs/dev/datastream_execution_mode.md

```diff
@@ -0,0 +1,377 @@
+enough to be able to deal with continuous data streams.
+
+One obvious outlier case is when you want to use a bounded job to bootstrap
+some job state that you then want to use in an unbounded job. For example, by
+running a bounded job using `STREAMING` mode, taking a savepoint, and then
+restoring that savepoint on an unbounded job. This is a very specific use case
+and one that might soon become obsolete when we allow producing a savepoint as
+additional output of a `BATCH` execution job.
+
+Another case where you might run a bounded job using `STREAMING` mode is when
+writing tests for code that will eventually run with unbounded sources. For
+testing it can be more natural to use a bounded source in those cases.
+
+## Configuring BATCH execution mode
+
+The execution mode can be configured via the `execution.runtime-mode` setting.
+There are three possible values:
+
+ - `STREAMING`: The classic DataStream execution mode (default)
+ - `BATCH`: Batch-style execution on the DataStream API
+ - `AUTOMATIC`: Let the system decide based on the boundedness of the sources
+
+This can be configured via command line parameters of `bin/flink run ...`, or
+programmatically when creating/configuring the `StreamExecutionEnvironment`.
+
+Here's how you can configure the execution mode via the command line:
+
+```bash
+$ bin/flink run -Dexecution.runtime-mode=BATCH examples/streaming/WordCount.jar
+```
+
+This example shows how you can configure the execution mode in code:
+
+ ```java
+Configuration config = new Configuration();
+config.set(ExecutionOptions.RUNTIME_MODE, RuntimeExecutionMode.BATCH);
+StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+ ```
+
+## Execution Behavior
+
+This section provides an overview of the execution behavior of `BATCH`
+execution mode and contrasts it with `STREAMING` execution mode. For more
+details, please refer to the FLIPs that introduced this feature:
+[FLIP-134](https://cwiki.apache.org/confluence/x/4i94CQ) and
+[FLIP-140](https://cwiki.apache.org/confluence/x/kDh4CQ). As well as the
+documentation about [task scheduling (TODO)](<TODO>).
+
+### Task Scheduling And Network Shuffle
+
+Flink jobs consist of different operations that are connected together in a
+dataflow graph. The system decides how to schedule the execution of these
+operations on different processes/machines (TaskManager) and how data is
+shuffled (sent) between them.
+
+Multiple operations/operators can be chained together using a feature called
+[chaining]({% link dev/stream/operators/index.md
+%}#task-chaining-and-resource-groups). A group of one or multiple (chained)
+operators that Flink considers as a unit of scheduling is called a _task_.
+Often the term _subtask_ is used to refer to the individual instances of tasks
+that are running in parallel on multiple TaskManagers but we will only use the
+term _task_ here.
+
+Task scheduling and the network shuffle work differently for `BATCH` execution
```

Comment:
"Task scheduling and the network shuffle" -> "Task scheduling and network shuffles"

"differently for `BATCH` execution mode and `STREAMING` execution mode" -> "differently for `BATCH` and `STREAMING` execution mode"

Reply:
fixing